dotnet-desktop-guide/winforms/migration/snippets/clipboard-dataobject-net10/net/csharp/ObsoletePatterns.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Windows.Forms;
+
+namespace ClipboardExamples
+{
+    // Examples of obsolete patterns that no longer work in .NET 10
+    public class ObsoletePatterns
+    {
+        // <ObsoleteCustomType>
+        [Serializable]
+        public class Person
+        {
+            public string Name { get; set; }
+            public int Age { get; set; }
+        }
+
+        public static void BrokenCustomTypeExample()
+        {
+            // This worked in .NET 8 and earlier but silently fails starting with .NET 9
```

Comment:
You should see a first chance exception under the debugger. The failure happens when OLE tries to make a copy of the data because of the default is `copy = true` on `SetData`. If it is set to `false` it will just work in process as we don't actually have to serialize the type.